packages/hash/frontend/src/components/SubmitButtonWrapper.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  faCircleExclamation,
+  faWarning,
+} from "@fortawesome/free-solid-svg-icons";
+import {
+  Box,
+  BoxProps,
+  Typography,
+  Tooltip,
+  tooltipClasses,
+  TooltipProps,
+  styled,
+  Collapse,
+} from "@mui/material";
+import React, { FC, forwardRef } from "react";
+import { Button } from "./Button";
+import { FontAwesomeIcon } from "./icons";
+
+const DisabledTooltip = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} classes={{ popper: className }} />
+))(({ theme }) => ({
+  [`& .${tooltipClasses.tooltip}`]: {
+    width: 260,
+    backgroundColor: theme.palette.common.white,
+    padding: "17px 16px",
+    color: theme.palette.orange[80],
+    boxShadow: theme.boxShadows.md,
+    border: `1px solid ${theme.palette.gray[20]}`,
+  },
+}));
+
+export type SubmitButtonWrapperProps = {
+  tooltip?: boolean;
+  helperText: string;
+} & BoxProps;
+
+export const SubmitButtonWrapper: FC<SubmitButtonWrapperProps> = forwardRef(
+  ({ children, tooltip, helperText, ...props }, ref) => {
+    if (process.env.NODE_ENV !== "production") {
+      if (!React.isValidElement(children) || children.type !== Button) {
+        throw new Error(
+          "Please wrap over a Button component => <ButtonHelperTextWrapper><Button /></ButtonHelperTextWrapper>",
+        );
+      }
+    }
+
+    if (tooltip && helperText) {
```

Comment:
Didn't consider that case initially. I don't think the ui should be rendered (tooltip or warning icon + helper text in collapse) if helperText="". I assume we will have scenarios where we assign helperText a value, then reset the value to an empty string. In that case, it might not make sense to show the tooltip/collapse with a warning icon and no text.

What do you think?

Reply:
Because we can't enforce `helperText !== ""` using TS, I think the component should be able to handle it. I agree that in that case it makes sense to show nothing at all, so why don't we just do this:
```TypeScript
 if (helperText === "") {
    return (
      <Box ref={ref} {...props}>
        {children}
      </Box>
    );
  }
```

That way neither a collapse or a tooltip gets rendered.

Reply:
Because the `Collapse` component depends on `Boolean(helperText)` changing, my suggestion won't work - resolving this as leaving this as is won't have any UI consequences.